### ISSUE TITLE
[Feature Request] Add lr_scheduler for full_finetune (single_device/distributed)

### ISSUE THREAD

**Issue by andyl98** [pytorch/torchtune#1060]
The scheduler inside yaml configs currently only gets picked up if the training job is with `LoRA`. I think we should adding a similar logic for full finetuning.
```    
# Learning rate scheduler can only be set up after number of steps
# has been computed
self._lr_scheduler = self._setup_lr_scheduler(
    cfg_lr_scheduler=cfg.lr_scheduler,
    num_training_steps=self.total_epochs * self._steps_per_epoch,
    last_epoch=self.global_step - 1,
)
```
(https://github.com/pytorch/torchtune/blob/f9cb9e6a94a9fec51edda65e5813db66419af3b1/recipes/lora_finetune_distributed.py#L254C1-L260C10)

**Comment by kartikayk** [comment]
Thanks so much for opening this issue!

I think the primary aspect we'd need to think a bit about is how the LR Scheduler interacts with the memory optimizations we have in place for single device fine-tuning. My initial thought is this shouldn't be too much of an issue for distributed (including 70B), though @ebsmothers can comment more intelligently on this.

BTW @RdoubleA, how would LR schedulers work with sample packing?

**Comment by RdoubleA** [comment]
LR scheduler is being added to full_finetune_single_device here: #1350. Distributed is still an open request (#1308, #952)

> BTW @RdoubleA, how would LR schedulers work with sample packing?

max steps per epoch is already adjusted when sample packing, so it's up to the user to adjust warmup / LR to account for seeing more samples within the same batch size. Although it would be good to provide guidance in our documentation on this.
